net-utils/src/sockets.rs

```diff
@@ -217,9 +217,19 @@ pub fn bind_in_range_with_config(
 }
 
 #[deprecated(since = "3.0.0", note = "Please bind to specific ports instead")]
+#[allow(deprecated)]
 pub fn bind_with_any_port_with_config(
     ip_addr: IpAddr,
     config: SocketConfiguration,
+) -> io::Result<UdpSocket> {
+    _bind_with_any_port_with_config(ip_addr, config)
+}
+
+// this private method works around a cargo bug involving nested deprecations
+// remove it with the above deprecated public method
+fn _bind_with_any_port_with_config(
+    ip_addr: IpAddr,
+    config: SocketConfiguration,
```

Comment:
stupid number two. this can be immediately removed in a followup in master, just isn't backportable. the private method is for scope-limiting